pkg/scheduler/scheduler.go

```diff
@@ -613,7 +613,9 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 	// Synchronously attempt to find a fit for the pod.
 	start := time.Now()
 	state := framework.NewCycleState()
-	scheduleResult, err := sched.Algorithm.Schedule(ctx, state, pod)
+	schedulingCycleCtx, cancel := context.WithCancel(ctx)
```

Comment:
is it better to keep using `ctx` instead of `schedulingCycleCtx` for reducing code changes?  
```go
ctx, cancel := context.WithCancel(ctx)
```

Reply:
After schedule cycle finishes, it will call `cancel`.  We are running plugins of binding cycle in another goroutine, they have not finished when schedule cycle finishes. If we use this new `ctx`(or create a new context that uses this new `ctx` as parent context) in binding cycle, it will be also cancelled even it has not finished. So we need keep original `ctx` to be used by binding cycle.